intelmq/lib/utils.py

```diff
@@ -808,3 +812,51 @@ def file_name_from_response(response: requests.Response) -> str:
     except KeyError:
         file_name = response.url.split("/")[-1]
     return file_name
+
+
+def list_all_bots() -> str:
+    mod = importlib.import_module('intelmq.bots')
+    dirs = ['collectors', 'parsers', 'experts', 'outputs']
+    allowed_types = ['bool', 'NoneType', 'str', 'int']
+    bot_types = [
+        'CollectorBot',
+        'ParserBot',
+        'OutputBot',
+        'Bot'
+    ]
+    root = mod.__path__[0]
+    root_dict = {}
+
+    for d in dirs:
+        root_dict[d.capitalize()] = {}
+        for dirs in os.listdir(os.path.join(root, d)):
+            bot = Path(os.path.join(root, d, dirs))
+            if bot.is_dir() and "__" not in bot.name:
+                for files in os.listdir(bot):
+                    f = Path(files)
+
+                    if f.suffix == '.py' and '__' not in f.name:
+                        mod = importlib.import_module("intelmq.bots.{}.{}.{}".format(d, bot.parts[-1:][0], f.name[0:-3]))
```

Comment:
If you want the filename from a `Path` object you can use `stem` - no need to manually slice the string.
Also isn't `d` already part of the `bot` path object- why use `d` directly in the format string instead of using the bot part?